src/compiler/c.rs

```diff
@@ -257,30 +261,33 @@ impl<T, I> CompilerHasher<T> for CCompilerHasher<I>
                    parsed_args.output_pretty(),
                    preprocessor_result.stdout.len());
 
-            let key = {
-                hash_key(&executable_digest,
-                         parsed_args.language,
-                         &parsed_args.common_args,
-                         &env_vars,
-                         &preprocessor_result.stdout)
-            };
-            // A compiler binary may be a symlink to another and so has the same digest, but that means
-            // the toolchain will not contain the correct path to invoke the compiler! Add the compiler
-            // executable path to try and prevent this
-            let weak_toolchain_key = format!("{}-{}", executable.to_string_lossy(), executable_digest);
-            Ok(HashResult {
-                key: key,
-                compilation: Box::new(CCompilation {
-                    parsed_args: parsed_args,
-                    #[cfg(feature = "dist-client")]
-                    preprocessed_input: preprocessor_result.stdout,
-                    executable: executable,
-                    compiler: compiler,
-                    cwd,
-                    env_vars,
-                }),
-                weak_toolchain_key,
-            })
+            Box::new(extra_hashes.and_then(move |extra_hashes| {
```

Comment:
You might not actually need the `Box::new` here (since this isn't the return value of the function, it's just the return value of this *closure*), but it's not really harmful either.